exonum-java-binding/common/src/test/java/com/exonum/binding/common/serialization/json/TransactionMessageJsonSerializerTest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.common.serialization.json;
+
+import static com.exonum.binding.common.crypto.CryptoFunctions.ed25519;
+import static com.exonum.binding.common.serialization.json.JsonSerializer.json;
+import static com.exonum.binding.test.Bytes.bytes;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+import com.exonum.binding.common.crypto.KeyPair;
+import com.exonum.binding.common.message.TransactionMessage;
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class TransactionMessageJsonSerializerTest {
+
+  @ParameterizedTest
+  @MethodSource("source")
+  void roundTripTest(TransactionMessage msg) {
+    String json = json().toJson(msg);
```

Comment:
This test relies on the fact that JsonSerializer includes a TransactionMessage adapter, correct? Shan't we use Gson here directly? Or move this test to JsonSerializerTest? I'd prefer to keep it here, separate.